alternator/executor.cc

```diff
@@ -759,6 +759,29 @@ future<executor::request_return_type> executor::tag_resource(client_state& clien
         co_return api_error::validation("The number of tags must be at least 1") ;
     }
     update_tags_map(*tags, tags_map,  update_tags_action::add_tags);
+
+    if (auto it = tags_map.find("system:default_time_to_live"); it != tags_map.end()) {
+        const int32_t default_ttl = std::atoi(it->second.data());
+        if (default_ttl <= 0) {
+            throw api_error::validation(
+                    format("Incorrect default_time_to_live tag value: {}. It must be a positive integer", it->second));
+        }
+        elogger.debug("Updating default_time_to_live of table {} to {}", schema->cf_name(), default_ttl);
+        co_await _mm.container().invoke_on(0, [default_ttl, global_schema = global_schema_ptr(schema)] (service::migration_manager& mm) mutable -> future<> {
+            schema_ptr prev_schema = global_schema.get();
+            // FIXME: the following needs to be in a loop. If mm.announce() below
+            // fails, we need to retry the whole thing.
+            auto group0_guard = co_await mm.start_group0_operation();
+
+            schema_builder builder(prev_schema);
+            builder.set_default_time_to_live(std::chrono::seconds(default_ttl));
+            auto schema = builder.build();
+
+            auto m = co_await mm.prepare_column_family_update_announcement(schema, false,  std::vector<view_ptr>(), group0_guard.write_timestamp());
+            co_await mm.announce(std::move(m), std::move(group0_guard));
```

Comment:
Hmm, I guess we can live with this for now (especially since we already have a FIXME here and in db::update_tags) but the right thing isn't to do the change of the schema and the change separately, but rather to prepare one mutation doing both, and then run it together.

Reply:
btw, this isn't needed, I can just pass an updated schema pointer straight to db::update_tags and it will be applied